File: OrangeJuice.Server.Core/Web/HttpClientAdapter.cs
using System;
using System.Net.Http;
using System.Threading.Tasks;

namespace OrangeJuice.Server.Web
{
	public class HttpClientAdapter : IHttpClient
	{
		#region Fields

		private readonly HttpClient _httpClient;

		#endregion

		#region Ctor
		public HttpClientAdapter(HttpClient httpClient)
		{
			_httpClient = httpClient;
		}

		#endregion

		#region IHttpClient members
		public Task<string> GetStringAsync(Uri url)
		{
			return _httpClient.GetStringAsync(url);
		}
		#endregion
	}
}

File: OrangeJuice.Server.Core.FSharp/Web/HttpClientAdapter.fs
namespace OrangeJuice.Server.FSharp.Web

open System
open System.Net.Http
open System.Threading.Tasks

open OrangeJuice.Server.Web

type HttpClientAdapter(httpClient : HttpClient) =
    interface IHttpClient with
        member this.GetStringAsync(url : Uri) : Task<string> =
            httpClient.GetStringAsync(url)
